libasherah.go

```diff
@@ -113,45 +148,73 @@ func Setup(kmsTypePtr unsafe.Pointer, metastorePtr unsafe.Pointer, rdbmsConnecti
 		globalDebugOutput = NullDebugOutput
 	}
 
-	setupAsherah(kmsType, metastore, rdbmsConnectionString, dynamoDbEndpoint, dynamoDbRegion, dynamoDbTableName,
-		enableRegionSuffix, serviceName, productId, preferredRegion, regionMapStr, verbose, sessionCache)
-
-	return ERR_NONE
-}
-
-func setupAsherah(kmsType string, metaStore string, rdbmsConnectionString string, dynamoDbEndpoint string,
-	dynamoDbRegion string, dynamoDbTableName string, enableRegionSuffix bool, serviceName string, productId string,
-	preferredRegion string, regionMapStr string, verbose bool, sessionCache bool) {
 	options := &Options{}
-	options.KMS = kmsType             // "kms"
-	options.ServiceName = serviceName // "chatterbox"
-	options.ProductID = productId     //"facebook"
+	options.KMS = kmsType
+	options.ServiceName = serviceName
+	options.ProductID = productId
 	options.Verbose = verbose
-	options.EnableSessionCaching = sessionCache
-	options.Metastore = metaStore //"dynamodb"
-	crypto := aead.NewAES256GCM()
+	options.EnableSessionCaching = enableSessionCaching
+	options.Metastore = metastore
 	options.ConnectionString = rdbmsConnectionString
 	options.DynamoDBEndpoint = dynamoDbEndpoint
 	options.DynamoDBRegion = dynamoDbRegion
 	options.DynamoDBTableName = dynamoDbTableName
 	options.EnableRegionSuffix = enableRegionSuffix
 	options.PreferredRegion = preferredRegion
-
 	if len(regionMapStr) > 0 {
-		regionMap := make(map[string]string)
-		pairs := strings.Split(regionMapStr, ",")
-		for _, pair := range pairs {
-			parts := strings.Split(pair, "=")
-			if len(parts) != 2 || len(parts[1]) == 0 {
-				panic("argument must be in the form of REGION1=ARN1[,REGION2=ARN2]")
-			}
-			region, arn := parts[0], parts[1]
-			regionMap[region] = arn
+		options.RegionMap = buildRegionMap(regionMapStr)
+	}
+	if len(expireAfterStr) > 0 {
+		expireAfter, err := time.ParseDuration(expireAfterStr)
+		if err != nil {
+			panic(err)
+		}
+		options.ExpireAfter = expireAfter
+	}
+	if len(checkIntervalStr) > 0 {
+		checkInterval, err := time.ParseDuration(checkIntervalStr)
+		if err != nil {
+			panic(err)
+		}
+		options.CheckInterval = checkInterval
+	}
+	if len(sessionCacheDurationStr) > 0 {
+		sessionCacheDuration, err := time.ParseDuration(sessionCacheDurationStr)
+		if err != nil {
+			panic(err)
 		}
+		options.SessionCacheDuration = sessionCacheDuration
+	}
+	if len(replicationReadConsistency) > 0 {
+		options.ReplicaReadConsistency = replicationReadConsistency
+	}
+	if sessionCacheMaxSize > 0 {
+		options.SessionCacheMaxSize = int(sessionCacheMaxSize)
+	}
```

Comment:
I'm not sure if I'm handling `sessionCacheMaxSize` properly? It is an optional attribute represented in Go as `int32`, and from the consumer side a value o f0 is used for the special case to not set it. Should this be done through the buffer somehow or?

Reply:
We'd want to make int32 versions of this, and then use 0 as the special case value:
```
export function int64_to_buffer(number: number): Buffer {
  const buffer: Buffer = Buffer.allocUnsafe(64 / 8)
  buffer.writeBigInt64LE(BigInt(number), 0)
  return buffer
}

export function buffer_to_int64(buffer: Buffer): string | number {
  return buffer.readInt64LE(0)
}
```